mondey_backend/src/mondey_backend/routers/scores.py

```diff
@@ -0,0 +1,242 @@
+            return TrafficLight.yellowWithCaveat.value
+        return TrafficLight.yellow.value
+    else:
+        if min_score is not None and min_score < lim_upper:
+            return TrafficLight.greenWithCaveat.value
+        return TrafficLight.green.value
+
+
+def compute_milestonegroup_feedback_summary(
+    session: SessionDep, child_id: int, answersession_id: int
+) -> dict[int, int]:
+    """
+    Compute the summary milestonegroup feedback for a single milestonegroup. This is done
+    by first calculating the mean score over all milestones that belong to the milestonegroup that
+    are relevant for the child when the given answersession was created. The mean is then
+    compared against the mean and standard deviation over the known population of children for the child's age.
+    When the statistics is outdated (older than a week currently) or there is none, it is recomputed and updated in the database.
+    See `compute_feedback_simple` for the feedback logic.
+
+    Parameters
+    ----------
+    session : SessionDep
+        database session
+    child_id : int
+        child to compute feedback for. Needed for age computation
+    answersession_id : int
+        answersession to compute feedback for. This contains the answers on which basis the feedback is computed
+
+    Returns
+    -------
+    dict[int, int]
+        Dictionary of milestonegroup_id -> feedback
+    """
+    answersession = session.get(MilestoneAnswerSession, answersession_id)
+
+    if answersession is None:
+        raise ValueError("No answersession with id: ", answersession_id)
+
+    # get child age
+    child = session.get(Child, child_id)
+    if child is None:
+        raise ValueError("No child with id: ", child_id)
+    age = get_child_age_in_months(child, answersession.created_at)
+
+    # extract milestonegroups
+    groups = set(answer.milestone_group_id for answer in answersession.answers.values())
+    today = datetime.now()
+
+    # for each milestonegroup, get the statistics, compute the current mean, and compute the feedback
+    # if the statistics is older than a week, we update it with the current data
+    feedback: dict[int, int] = {}
+    for group in groups:
+        # print("current group: ", group)
+        stats = session.exec(
+            select(MilestoneGroupAgeScoreCollection).where(
+                MilestoneGroupAgeScoreCollection.milestone_group_id == group
+            )
+        ).first()
+        # if stats is not None:
+
+        if stats is None or stats.created_at < today - timedelta(days=7):
+            new_stats = calculate_milestonegroup_statistics_by_age(session, group)
+
+            if new_stats is None:
+                raise ValueError("No statistics for milestone group: ", group)
+
+            # update stuff in database
+            for new_score in new_stats.scores:
+                session.merge(new_score)
+
+            session.merge(new_stats)
```

Comment:
do you know if the merge is recursive? unclear to me if this duplicates the merges done above or not, although I guess either way it's ok